app/forms/reset_password_form.rb

```diff
@@ -56,6 +58,21 @@ def mark_profile_inactive
     user.proofing_component&.destroy
   end
 
+  # It is possible for an account that is resetting their password to be "invalid".
+  # If an unconfirmed account (which must have one unconfirmed email address) resets their
+  # password and a different account then adds and confirms that same email address,
+  # the initial account is no longer able to confirm their email address and is effectively invalid.
+  #
+  # They may still have a valid forgot password link for the initial account, which would normally
+  # mark their email as confirmed when they set a new password, but we do not want to allow it
+  # because we only allow an email address to be confirmed on one account.
+  def invalid_account?
+    !user.confirmed? &&
+      EmailAddress.confirmed.exists?(
+        email_fingerprint: user.email_addresses.map(&:email_fingerprint),
+      )
+  end
```

Comment:
should we add a unit test for this method/failure?

Reply:
I'm comfortable with the test in https://github.com/18F/identity-idp/pull/6042/files#diff-6169030eddc977e1ae39fe9c01f51f38bf4c8dfbec994c3539a8c10d5340b8caR154-R169 covering the expected behavior. Is there an aspect or edge case missing?

Reply:
I think for odd model joins like this, that doing a unit tests is clearer than an acceptance test, but I don't feel that strongly, the overall behavior is covered by the one you linked

Reply:
If it helps, there isn't a join and could probably be simplified to just be `user.email_addresses.first` since an unconfirmed account should only ever have one email address